### PR TITLE
Component: Contract Box component

### PR DESCRIPTION
**Pull request by blackdevelopa**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

**Screenshots/Recordings**
Screenshot 2022-09-30 at 00 27 08
Screenshot 2022-09-30 at 00 27 23

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses https://app.zenhub.com/workspaces/metamask-mobile-5f984938ddc0e4001d4b79cb/issues/metamask/metamask-planning/184

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/component-library/components-temp/Contracts/ContractBoxBase/ContractBoxBase.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import Icon, { IconName, IconSize } from '../../../components/Icon';
+import styleSheet from './ContractBoxBase.styles';
+import AvatarToken from '../../../components/Avatars/AvatarToken';
+import {
+  EXPORT_ICON_TEST_ID,
+  COPY_ICON_TEST_ID,
+  CONTRACT_BOX_TEST_ID,
+  CONTRACT_BOX_NO_PET_NAME_TEST_ID,
+} from './ContractBoxBase.constants';
+import { useStyles } from '../../../hooks';
```
Comment (jpcloureiro) [inline app/component-library/components-temp/Contracts/ContractBoxBase/ContractBoxBase.tsx:15]: Sort imports from top to botttom:
1 -  imports from external packages
2 - import from outside component filter
3 - imports from the current folder

**Review by jpcloureiro — CHANGES_REQUESTED** [review]
~~If the component has not been fully standardised & reviewed by the Design System team then I would move it inside the component-temp folder~~

Nevermind the comment, it is already on the temp folder 👍

File: app/component-library/components-temp/Contracts/ContractBoxBase/ContractBoxBase.tsx (hunk 1 above)
Comment (blackdevelopa) [reply app/component-library/components-temp/Contracts/ContractBoxBase/ContractBoxBase.tsx]: Thanks. Updated now

File: app/component-library/components-temp/Contracts/ContractBoxBase/ContractBoxBase.types.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  contractPetName?: string;
+  contractLocalImage: ImageSourcePropType;
+  /**
+   * function that copies the contract address to the clipboard
+   */
+  handleCopyAddress: () => void;
+  /**
+   * function that opens contract in block explorer
+   */
+  handleExportAddress: () => void;
```
[the remark this answers is not in the document]
Comment (blackdevelopa) [reply app/component-library/components-temp/Contracts/ContractBoxBase/ContractBoxBase.types.ts]: Thanks. Updated

**Review by jpcloureiro — APPROVED** [review]
lgtm!

**Pull request merged**